js-src/Settings.ts

```diff
@@ -0,0 +1,210 @@
+ */
+export function loadC2paSettings(json: string): void {
+  neon.loadSettings(json);
+}
+
+/**
+ * Load settings from a TOML string and apply them globally.
+ * @param toml The TOML string containing the settings configuration
+ */
+export function loadC2paSettingsToml(toml: string): void {
+  neon.loadSettingsToml(toml);
+}
+
+/**
+ * Load settings from a TOML or JSON file path and apply them globally.
+ * The file format is determined by the file extension (.toml for TOML, otherwise JSON).
+ * @param filePath The path to the settings file
+ */
+export async function loadSettingsFromFile(filePath: string): Promise<void> {
+  const ext = path.extname(filePath).toLowerCase();
+  const content = await fs.readFile(filePath, "utf8");
+  if (ext === ".toml") {
+    loadC2paSettingsToml(content);
+    return;
+  }
+  // Assume JSON (or JSON5 that is valid JSON) otherwise
+  // If needed, callers can ensure the content is valid JSON.
+  loadC2paSettings(content);
+}
+
+/**
+ * Load settings from a URL and apply them globally.
+ * The format is determined by the content-type header or file extension (.toml for TOML, otherwise JSON).
+ * @param url The URL to fetch the settings from
+ */
+export async function loadSettingsFromUrl(url: string): Promise<void> {
+  const res = await fetch(url);
+  if (!res.ok) {
+    throw new Error(
+      `Failed to fetch settings from URL: ${res.status} ${res.statusText}`,
+    );
+  }
+  const contentType = res.headers.get("content-type") || "";
+  const text = await res.text();
+
+  const isToml =
+    contentType.includes("toml") || url.toLowerCase().endsWith(".toml");
+  if (isToml) {
+    loadC2paSettingsToml(text);
+    return;
+  }
+  // Treat as JSON by default
+  loadC2paSettings(text);
+}
+
+/**
+ * Get the current settings as a JSON string.
+ * @returns The JSON representation of the current settings
+ */
+export function getSettingsJson(): string {
+  return neon.getSettingsJson();
+}
+
+/**
+ * Load trust configuration into the main trust settings.
+ * @param trustConfig The trust configuration object
+ */
+export function loadTrustConfig(trustConfig: TrustConfig): void {
+  // Convert camelCase to snake_case for Rust compatibility
+  // Explicitly set undefined fields to null to ensure they are cleared
+  const rustConfig = {
+    verify_trust_list: trustConfig.verifyTrustList,
+    user_anchors: trustConfig.userAnchors ?? null,
+    trust_anchors: trustConfig.trustAnchors ?? null,
+    trust_config: trustConfig.trustConfig ?? null,
+    allowed_list: trustConfig.allowedList ?? null,
+  };
+  neon.loadTrustConfig(JSON.stringify(rustConfig));
+}
+
+/**
+ * Load trust configuration into the CAWG trust settings.
+ * @param trustConfig The trust configuration object
+ */
+export function loadCawgTrustConfig(trustConfig: TrustConfig): void {
+  // Convert camelCase to snake_case for Rust compatibility
+  // Explicitly set undefined fields to null to ensure they are cleared
+  const rustConfig = {
```

Comment:
`rustConfig` because it's related to the Rust codebase, or type with the missing `t` for `trustconfig`?